src/app/components/FeeSelectField/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Form, Alert, Slider } from 'antd';
+import { AppState } from 'store/reducers';
+import Loader from 'components/Loader';
+import { SliderValue } from 'antd/lib/slider';
+import { CHAIN_TYPE } from 'utils/constants';
+import { getOnChainFeeEstimates } from 'modules/payment/actions';
+import { getNodeChain } from 'modules/node/selectors';
+import './style.less';
+
+interface StateProps {
+  onChainFees: AppState['payment']['onChainFees'];
+  feesError: AppState['payment']['feesError'];
+  isFetchingFees: AppState['payment']['isFetchingFees'];
+  chain: ReturnType<typeof getNodeChain>;
+}
+
+interface DispatchProps {
+  getOnChainFeeEstimates: typeof getOnChainFeeEstimates;
+}
+
+interface OwnProps {
+  showFeeMsg?: boolean;
+  onChange(fee: number): void;
+}
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+interface State {
+  fee: number | undefined,
+}
+
+class FeeSelectField extends Component<Props, State> {
+  state: State = {
+    fee: 0,
+  }
+
+  componentDidMount() {
+    this.props.getOnChainFeeEstimates();
+  }
+
+  componentWillReceiveProps(nextProps: Props) {	
+    const { onChainFees } = this.props;	
+    if (nextProps.onChainFees !== onChainFees && nextProps.onChainFees !== null) {	
+      const { fastestFee } = nextProps.onChainFees;
+      this.setState({ fee: fastestFee });	
```

Comment:
Also just took a look at the parent component and am wondering whether you wouldn’t need local state in this component and you can just pass the `fee` as a prop from the `ChainSend` component since it seems like you’re managing changes to both this local state and the parents local state?

All good if you have your reasoning that I don’t see, but in any case these are just architecture suggestions and not functionally critical. 

Love the slider UI btw 👍👍👍